# London Airbnb Analysis — SQL Analysis

This notebook loads the cleaned listings data into a SQLite database and uses SQL queries to answer questions about London Airbnb listings.

In [17]:
# Import libraries
import pandas as pd
import sqlite3

print("Libraries loaded!")

Libraries loaded!


## 1. Load Data and Create Database

We load the cleaned listings into a SQLite database so we can query it with SQL.

In [18]:
# Load the cleaned listings data
listings = pd.read_csv('../data/listings_clean.csv')

print("Listings loaded!")
print("Rows:", len(listings))

Listings loaded!
Rows: 60951


In [19]:
# Create a SQLite database
conn = sqlite3.connect(':memory:')

# Load listings into the database as a table
listings.to_sql('listings', conn, if_exists='replace', index=False)

print("Database ready!")

Database ready!


## 2. Simple Queries

Starting with basic SQL to understand the data.

In [20]:
# Select all columns from the first 5 rows
query = "SELECT * FROM listings LIMIT 5"
result = pd.read_sql(query, conn)
print(result)

        id                         listing_url     scrape_id last_scraped  \
0  13913.0  https://www.airbnb.com/rooms/13913  2.025091e+13   2025-09-16   
1  15400.0  https://www.airbnb.com/rooms/15400  2.025091e+13   2025-09-16   
2  17402.0  https://www.airbnb.com/rooms/17402  2.025091e+13   2025-09-16   
3  36274.0  https://www.airbnb.com/rooms/36274  2.025091e+13   2025-09-15   
4  36299.0  https://www.airbnb.com/rooms/36299  2.025091e+13   2025-09-15   

        source                                               name  \
0  city scrape                Holiday London DB Room Let-on going   
1  city scrape                Bright Chelsea  Apartment. Chelsea!   
2  city scrape   Very Central Modern 3-Bed/2 Bath By Oxford St W1   
3  city scrape  Bright 1 bedroom apt off brick lane in Shoreditch   
4  city scrape                Kew Gardens 3BR house in cul-de-sac   

                                         description  \
0  My bright double bedroom with a large window h...   
1  Lots of

In [21]:
# Select just the name, neighbourhood and price
query = "SELECT name, neighbourhood_cleansed, price FROM listings LIMIT 10"
result = pd.read_sql(query, conn)
print(result)

                                                name  neighbourhood_cleansed  \
0                Holiday London DB Room Let-on going               Islington   
1                Bright Chelsea  Apartment. Chelsea!  Kensington and Chelsea   
2   Very Central Modern 3-Bed/2 Bath By Oxford St W1             Westminster   
3  Bright 1 bedroom apt off brick lane in Shoreditch           Tower Hamlets   
4                Kew Gardens 3BR house in cul-de-sac    Richmond upon Thames   
5                    You are GUARANTEED to love this                Haringey   
6         SUNNY ROOM PRIVATE BATHROOM PLUS BREAKFAST  Hammersmith and Fulham   
7                                    Short Term Home  Hammersmith and Fulham   
8           SPACIOUS ROOM IN CONTEMPORARY STYLE FLAT               Southwark   
9  2 Double bed apartment in quiet area North London                  Barnet   

   price  
0   70.0  
1  149.0  
2  411.0  
3  210.0  
4  280.0  
5   90.0  
6   61.0  
7  340.0  
8   49.0  
9  213.0 

In [22]:
# How many listings are there in total?
query = "SELECT COUNT(*) as total_listings FROM listings"
result = pd.read_sql(query, conn)
print(result)

   total_listings
0           60951


In [23]:
# What room types exist?
query = "SELECT DISTINCT room_type FROM listings"
result = pd.read_sql(query, conn)
print(result)

         room_type
0     Private room
1  Entire home/apt
2       Hotel room
3      Shared room


## 3. WHERE Queries

Filter the data to find specific listings.

In [24]:
# Listings under £50 per night
query = "SELECT name, neighbourhood_cleansed, price FROM listings WHERE price < 50 LIMIT 10"
result = pd.read_sql(query, conn)
print(result)

                                                name  neighbourhood_cleansed  \
0           SPACIOUS ROOM IN CONTEMPORARY STYLE FLAT               Southwark   
1                   Quiet Comfortable Room in Fulham  Hammersmith and Fulham   
2  NICE FAMILY HOME opposite NATURAL LANDSCAPED PARK                  Barnet   
3   Double Room 2 Miles from Wimbledon in Morden SM4                  Merton   
4                        Spacious Bright Room in SE3               Greenwich   
5   Double Room (Unavailable for check in 9-11 July)                 Enfield   
6   A double Room 5mins from King's College Hospital                 Lambeth   
7                    Penthouse Living in East London    Barking and Dagenham   
8                  London Whitechapel Room with WIFI           Tower Hamlets   
9                LOVELY & COZY ROOM WITH KITCHENETTE                 Croydon   

   price  
0   49.0  
1   48.0  
2   40.0  
3   35.0  
4   43.0  
5   36.0  
6   42.0  
7   45.0  
8   44.0  
9   23.0 

In [25]:
# Only entire home listings
query = "SELECT name, neighbourhood_cleansed, price FROM listings WHERE room_type = 'Entire home/apt' LIMIT 10"
result = pd.read_sql(query, conn)
print(result)

                                                name  neighbourhood_cleansed  \
0                Bright Chelsea  Apartment. Chelsea!  Kensington and Chelsea   
1   Very Central Modern 3-Bed/2 Bath By Oxford St W1             Westminster   
2  Bright 1 bedroom apt off brick lane in Shoreditch           Tower Hamlets   
3                Kew Gardens 3BR house in cul-de-sac    Richmond upon Thames   
4                                    Short Term Home  Hammersmith and Fulham   
5  2 Double bed apartment in quiet area North London                  Barnet   
6       Cosy Double studio in Zone 2 Hammersmith (6)  Hammersmith and Fulham   
7       Cosy Double studio in Zone 2 Hammersmith (1)  Hammersmith and Fulham   
8       Cosy Double studio in Zone 2 Hammersmith (4)  Hammersmith and Fulham   
9                       Sunny studio with park views                 Hackney   

   price  
0  149.0  
1  411.0  
2  210.0  
3  280.0  
4  340.0  
5  213.0  
6   78.0  
7   78.0  
8   89.0  
9  159.0 

In [26]:
# Only superhost listings
query = "SELECT name, neighbourhood_cleansed, price FROM listings WHERE host_is_superhost = 't' LIMIT 10"
result = pd.read_sql(query, conn)
print(result)

                                                name  neighbourhood_cleansed  \
0                Holiday London DB Room Let-on going               Islington   
1   Very Central Modern 3-Bed/2 Bath By Oxford St W1             Westminster   
2                    You are GUARANTEED to love this                Haringey   
3         SUNNY ROOM PRIVATE BATHROOM PLUS BREAKFAST  Hammersmith and Fulham   
4                                    Short Term Home  Hammersmith and Fulham   
5           SPACIOUS ROOM IN CONTEMPORARY STYLE FLAT               Southwark   
6  Room with a view, shared flat,  central  Bankside               Southwark   
7                           You Will Save Money Here                  Barnet   
8                   Quiet Comfortable Room in Fulham  Hammersmith and Fulham   
9       Cosy Double studio in Zone 2 Hammersmith (6)  Hammersmith and Fulham   

   price  
0   70.0  
1  411.0  
2   90.0  
3   61.0  
4  340.0  
5   49.0  
6   96.0  
7   71.0  
8   48.0  
9   78.0 

## 4. ORDER BY Queries

Sort the data to find the cheapest and most expensive listings.

In [27]:
# 5 cheapest listings
query = "SELECT name, neighbourhood_cleansed, room_type, price FROM listings ORDER BY price ASC LIMIT 5"
result = pd.read_sql(query, conn)
print(result)

                                                name neighbourhood_cleansed  \
0                              shared room in London         Waltham Forest   
1                   Spacious Double bed Loft Fairlop              Redbridge   
2                         Luxury Urban Cabin retreat                 Sutton   
3                     Peace & Tranquil in Kennington                Lambeth   
4  Its oly 5-10 Mins walk to Croydon Shopping Centre                Croydon   

         room_type  price  
0      Shared room    7.0  
1     Private room    8.0  
2  Entire home/apt    9.0  
3     Private room   10.0  
4     Private room   10.0  


In [28]:
# 5 most expensive listings
query = "SELECT name, neighbourhood_cleansed, room_type, price FROM listings ORDER BY price DESC LIMIT 5"
result = pd.read_sql(query, conn)
print(result)

                                                name  neighbourhood_cleansed  \
0               Trendy 1 bed Paddington/Regents Park             Westminster   
1  Central and cosy with large balcony and sunlight!           Tower Hamlets   
2                          The Feel of London Living  Hammersmith and Fulham   
3     a huge house with lots of space and very clean             Westminster   
4           Stylish 2 Bedroom Maisonette in Victoria             Westminster   

         room_type  price  
0  Entire home/apt  999.0  
1  Entire home/apt  999.0  
2  Entire home/apt  999.0  
3  Entire home/apt  999.0  
4  Entire home/apt  999.0  


## 5. GROUP BY Queries

Group the data to find averages and counts by category.

In [29]:
# Average price per neighbourhood
query = """
SELECT neighbourhood_cleansed as neighbourhood,
       ROUND(AVG(price), 2) as avg_price,
       COUNT(*) as num_listings
FROM listings
GROUP BY neighbourhood_cleansed
ORDER BY avg_price DESC
LIMIT 10
"""
result = pd.read_sql(query, conn)
print(result)

            neighbourhood  avg_price  num_listings
0          City of London     276.42           408
1  Kensington and Chelsea     269.44          4570
2             Westminster     262.98          8116
3                  Camden     192.92          4255
4  Hammersmith and Fulham     178.63          2630
5              Wandsworth     174.97          2978
6               Islington     173.60          2734
7    Richmond upon Thames     171.57           802
8               Southwark     151.37          3194
9           Tower Hamlets     151.30          4222


In [30]:
# Count of listings by room type
query = """
SELECT room_type,
       COUNT(*) as num_listings,
       ROUND(AVG(price), 2) as avg_price
FROM listings
GROUP BY room_type
ORDER BY num_listings DESC
"""
result = pd.read_sql(query, conn)
print(result)

         room_type  num_listings  avg_price
0  Entire home/apt         41596     215.15
1     Private room         19117      79.96
2      Shared room           184      58.92
3       Hotel room            54     207.78


In [31]:
# Superhost vs regular host
query = """
SELECT host_is_superhost,
       COUNT(*) as num_listings,
       ROUND(AVG(price), 2) as avg_price,
       ROUND(AVG(review_scores_rating), 2) as avg_rating
FROM listings
GROUP BY host_is_superhost
"""
result = pd.read_sql(query, conn)
print(result)

  host_is_superhost  num_listings  avg_price  avg_rating
0              None          1324     175.56        4.65
1                 f         45308     168.04        4.60
2                 t         14319     185.34        4.85


## 6. Summary

Key findings from the SQL analysis based on 60,951 London Airbnb listings:

**Price by neighbourhood:**
- Most expensive borough: **City of London** at £276/night average (408 listings)
- Second most expensive: **Kensington and Chelsea** at £269/night (4,570 listings)
- **Westminster** third at £263/night with the most central listings (8,116)
- **Tower Hamlets** and **Southwark** are more affordable at ~£151/night

**Room types:**
- **Entire home/apt** dominates with 41,596 listings at an average of £215/night
- **Private rooms** are the budget option with 19,117 listings at £80/night
- **Hotel rooms** are rare (54 listings) but priced similarly to entire homes at £208/night
- **Shared rooms** are cheapest at £59/night but very uncommon (184 listings)

**Superhost vs regular host:**
- There are 14,319 superhosts vs 45,308 regular hosts
- Superhosts charge slightly more: **£185** vs **£168** for regular hosts
- Superhosts get significantly better ratings: **4.85** vs **4.60**
- This suggests superhost status is worth it — better service commands a premium

**Price extremes:**
- Cheapest listing: £7/night (shared room in Waltham Forest)
- Most expensive listings: £999/night (entire homes in Westminster and Tower Hamlets)

In [32]:
# Always close the database connection when done
conn.close()

print("Database connection closed!")

Database connection closed!
